src/gl/gl_builders.js

```diff
@@ -145,213 +145,149 @@ GLBuilders.buildPolylines = function (
-                    mmax = mid.length;
-                }
-                else {
-                    mid.push(line[line.length-1]); // use line end instead of last midpoint
-                    mmax = mid.length - 1;
-                }
-
-                // Make anchors by connecting midpoints to line joints
-                for (p=0; p < mmax; p++)  {
-                    anchors.push([mid[p], line[(p+1) % line.length], mid[(p+1) % mid.length]]);
-                }
-            }
-            else {
-                // Degenerate case, a 3-point line is just a single anchor
-                anchors = [[line[0], line[1], line[2]]];
-            }
-
-            for (p=0; p < anchors.length; p++) {
-                if (!remove_tile_edges) {
-                    buildAnchor(anchors[p][0], anchors[p][1], anchors[p][2]);
-                }
-                else {
-                    var edge1 = GLBuilders.isOnTileEdge(anchors[p][0], anchors[p][1]);
-                    var edge2 = GLBuilders.isOnTileEdge(anchors[p][1], anchors[p][2]);
-                    if (!edge1 && !edge2) {
-                        buildAnchor(anchors[p][0], anchors[p][1], anchors[p][2]);
-                    }
-                    else if (!edge1) {
-                        buildSegment(anchors[p][0], anchors[p][1]);
-                    }
-                    else if (!edge2) {
-                        buildSegment(anchors[p][1], anchors[p][2]);
-                    }
-                }
-            }
+    // Add to equidistant pairs of vertices
+    function addVertexPair ( coord, normal, v_pct ) {
+        if (scaling_index) {
+            //  a. If scaling is on add the vertex (the currCoord) and the scaling Vecs (normals pointing where to extrude the vertexes)
+            vertices.push(coord);
+            vertices.push(coord);
+            scalingVecs.push(normal);
+            scalingVecs.push(Vector.neg(normal));
+        } else {
+            //  b. Add the extruded vertexes
+            vertices.push([ coord[0] + normal[0] * halfWidth,
+                            coord[1] + normal[1] * halfWidth ]);
+            vertices.push([ coord[0] - normal[0] * halfWidth,
+                            coord[1] - normal[1] * halfWidth ]);
         }
-        // Single 2-point segment
-        else if (line.length === 2) {
-            buildSegment(line[0], line[1]); // TODO: replace buildSegment with a degenerate form of buildAnchor? buildSegment is still useful for debugging
+
+        // c) Add uv's if they are enable
+        if (texcoord_index) {
+            texcoords.push( [ max_u, (1-v_pct)*min_v + v_pct*max_v ],
+                            [ min_u, (1-v_pct)*min_v + v_pct*max_v ]);
         }
     }
 
-    // Add vertices to buffer
-    for (var v=0; v < vertices.length; v++) {
-        vertex_template[0] = vertices[v][0];
-        vertex_template[1] = vertices[v][1];
+    // Add a vertex based on the index position into the VBO 
+    function addIndex ( index ) {
```

Comment:
Not sure if this is important, but defining a function within another function means that on every function invocation we are going to be creating a new function object. 

Normally this is not a huge deal in JS, but you mentioned that some of this code is in a tight inner loop.